### ISSUE TITLE
ErrorException: unserialize()

### ISSUE THREAD

**Issue by theofidry** [FriendsOfPHP/PHP-CS-Fixer#1770]
When trying to run the fixer (`@2.0-dev` on OSX 10.9) I got the following error:

```
  [ErrorException]
  unserialize(): Error at offset 4009 of 4540 bytes

Exception trace:
 () at /myPath/.composer/vendor/fabpot/php-cs-fixer/src/FileCacheManager.php:127
 {closure}() at n/a:n/a
 unserialize() at /myPath/.composer/vendor/fabpot/php-cs-fixer/src/FileCacheManager.php:127
 PhpCsFixer\FileCacheManager->readFromFile() at /myPath/.composer/vendor/fabpot/php-cs-fixer/src/FileCacheManager.php:54
 PhpCsFixer\FileCacheManager->__construct() at /myPath/.composer/vendor/fabpot/php-cs-fixer/src/Fixer.php:119
 PhpCsFixer\Fixer->fix() at /myPath/.composer/vendor/fabpot/php-cs-fixer/src/Console/Command/FixCommand.php:354
 PhpCsFixer\Console\Command\FixCommand->execute() at /myPath/.composer/vendor/symfony/console/Command/Command.php:259
 Symfony\Component\Console\Command\Command->run() at /myPath/.composer/vendor/symfony/console/Application.php:844
 Symfony\Component\Console\Application->doRunCommand() at /myPath/.composer/vendor/symfony/console/Application.php:192
 Symfony\Component\Console\Application->doRun() at /myPath/.composer/vendor/symfony/console/Application.php:123
 Symfony\Component\Console\Application->run() at /Users/Theo/.composer/vendor/fabpot/php-cs-fixer/php-cs-fixer:44
```

As it comes from the `FileCacheManager` I suspected a cache issue and indeed deleting `.php_cs.cache` solved the problem. I'm still reporting this issue as I believe it should be handled a better way somehow (cache invalid? get rid of it for example).

**Comment by keradus** [comment]
Could you please provide the `.php_cs.cache` file that was revealed the issue?

**Comment by theofidry** [comment]
```
a:3:{s:7:"version";s:59:"2.0-DEV:dev-master#a674d597cf248dbfad467a2609f6a5cd2ff65e44";s:5:"rules";a:23:{s:8:"encoding";b:1;s:16:"full_opening_tag";b:1;s:26:"blank_line_after_namespace";b:1;s:6:"braces";b:1;s:16:"class_definition";b:1;s:6:"elseif";b:1;s:20:"function_declaration";b:1;s:19:"lowercase_constants";b:1;s:18:"lowercase_keywords";b:1;s:21:"method_argument_space";b:1;s:14:"no_closing_tag";b:1;s:29:"no_spaces_after_function_name";b:1;s:28:"no_spaces_inside_parenthesis";b:1;s:18:"no_tab_indentation";b:1;s:22:"no_trailing_whitespace";b:1;s:33:"no_trailing_whitespace_in_comment";b:1;s:24:"single_blank_line_at_eof";b:1;s:27:"single_import_per_statement";b:1;s:25:"single_line_after_imports";b:1;s:30:"switch_case_semicolon_to_colon";b:1;s:17:"switch_case_space";b:1;s:17:"unix_line_endings";b:1;s:19:"visibility_required";b:1;}s:6:"hashes";a:53:{s:26:"Exceptions/HandlerTest.php";i:2842065013;s:31:"Glyph/EmailSubjectGlyphTest.php";i:4001396064;s:50:"Handler/Appointment/Customer/AcceptHandlerTest.php";i:3358701551;s:50:"Handler/Appointment/Customer/CancelHandlerTest.php";i:2420453419;s:38:"Helpers/Client/FreegeoipClientTest.php";i:846916476;s:33:"Http/Middleware/AppLocaleTest.php";i:1634166797;s:35:"Http/Middleware/HostCheckerTest.php";i:2552780729;s:51:"Http/Payload/Input/AbstractInputPayloadTestCase.php";i:2167739070;s:77:"Http/Payload/Input/Appointment/CreateAppointmentOptionsRequestPayloadTest.php";i:2094608848;s:53:"Http/Payload/Input/Appointment/RequestPayloadTest.php";i:2680193521;s:48:"Http/Payload/Input/BookingRequestPayloadTest.php";i:44514586;s:52:"Http/Payload/Input/InputPayloadTestCaseInterface.php";i:1020340163;s:48:"Payload/Appointment/AcceptRequestPayloadTest.php";i:1455618701;s:48:"Payload/Appointment/CancelRequestPayloadTest.php";i:3592911816;s:38:"Prophecy/Argument/DateTimeArgument.php";i:2500912577;s:46:"Prophecy/Argument/Token/DifferentDateToken.php";i:554022444;s:53:"Prophecy/Argument/Token/DifferentValueObjectToken.php";i:1713126122;s:46:"Prophecy/Argument/Token/IdenticalDateToken.php";i:1021573110;s:53:"Prophecy/Argument/Token/IdenticalValueObjectToken.php";i:1166349290;s:41:"Prophecy/Argument/ValueObjectArgument.php";i:4267997771;s:42:"Serializer/DateTimestampSerializerTest.php";i:3067092361;s:55:"Services/Appointment/Customer/AcceptanceServiceTest.php";i:2447881237;s:57:"Services/Appointment/Customer/CancellationServiceTest.php";i:3108989121;s:12:"TestCase.php";i:3614415748;s:49:"Tests/Integration/Provider/LocaleProviderTest.php";i:3978763267;s:51:"Tests/Integration/Provider/NotifierProviderTest.php";i:2574818074;s:52:"Tests/Integration/Provider/SchedulerProviderTest.php";i:3118891154;s:52:"Tests/Unit/Exceptions/CastableExceptionTraitTest.php";i:280111270;s:67:"Tests/Unit/Locale/Extractor/DefaultLocaleExtractorDecoratorTest.php";i:899827269;s:49:"Tests/Unit/Locale/Extractor/HostExtractorTest.php";i:1864918791;s:52:"Tests/Unit/Locale/Extractor/RequestExtractorTest.php";i:2811945691;s:32:"Tests/Unit/Locale/LocaleTest.php";i:4116060836;s:51:"Tests/Unit/Locale/Provider/InMemoryProviderTest.php";i:4242694942;s:50:"Tests/Unit/Locale/Setter/ApplicationSetterTest.php";i:638414027;s:45:"Tests/Unit/Locale/Setter/CarbonSetterTest.php";i:3322720140;s:46:"Tests/Unit/Locale/Setter/RequestSetterTest.php";i:3660820769;s:47:"Tests/Unit/Locale/Setter/SetterRegistryTest.php";i:1752115805;s:45:"Tests/Unit/Locale/Setter/SystemSetterTest.php";i:2131750055;s:49:"Tests/Unit/Locale/Setter/TranslatorSetterTest.php";i:2267789286;s:51:"Tests/Unit/Locale/Setter/UrlGeneratorSetterTest.php";i:1246317426;s:51:"Tests/Unit/Manager/DateTime/HolidaysManagerTest.php";i:2900087846;s:66:"Tests/Unit/Scheduler/Appointment/CustomerReminderSchedulerTest.php";i:1035617139;s:63:"Tests/Unit/Scheduler/Appointment/SalonReminderSchedulerTest.php";i:2094361856;s:55:"Tests/Unit/ValueObject/DateTime/CarbonGeneratorTest.php";i:3488464221;s:56:"Tests/Unit/ValueObject/DateTime/DateTimeIntervalTest.php";i:2536646381;s:47:"Tests/Unit/ValueObject/DateTime/HolidaysTest.php";i:1826617722;s:55:"Tests/Unit/ValueObject/DateTime/SalonDayFactoryTest.php";i:2897743007;s:48:"Tests/Unit/ValueObject/DateTime/SalonDayTest.php";i:3401365676;s:57:"Tests/Unit/ValueObject/DateTime/SalonWeekIteratorTest.php";i:191504580;s:49:"Tests/Unit/ValueObject/DateTime/SalonWeekTest.php";i:3278268234;s:64:"Tests/Unit/ValueObject/DateTime/SalonWorkingDaysIteratorTest.php";i:705125096;s:50:"Tests/Unit/ValueObject/DateTime/WorkingDayTest.php";i:3938863716;s:44:"Tests/ValueObject/Http/HostWhitelistTest.php";i:1677210806;}}
```

**Comment by keradus** [comment]
Thanks!

I will play with it at night or tomorrow.

**Comment by theofidry** [comment]
Cheers

**Comment by keradus** [comment]
For some reason you have invalid serialized data, and I cannot guess why :/
You have:
`s:47:"Tests/Unit/ValueObject/DateTime/HolidaysTest.php"`
instead of:
`s:48:"Tests/Unit/ValueObject/DateTime/HolidaysTest.php"`

**Comment by theofidry** [comment]
I'm unsure of what may have cause this. But regardless of the cause, in this situation wouldn't it be better to disregard the cache and rebuild it rather than displaying this error?

**Comment by keradus** [comment]
Yeah, for sure, I'm just in the middle of working for patch for it (1.x first).

**Comment by keradus** [comment]
ref https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1775

**Comment by theofidry** [comment]
Thinking of it, it may be due to a refactoring done in PHPStorm (I'm sometimes lazy when it comes to cache files even if I usually disable them).

**Comment by keradus** [comment]
Closed due to merge of #1775

**Comment by theofidry** [comment]
Thank you @keradus :)
